src/api/firebase.js

```diff
@@ -115,3 +115,64 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export function comparePurchaseUrgency(shoppingList) {
+	// Loop though the shopping list and for each item and calculate days until next purchase and days since last purchase
+	// Store days until next purchase into property item.purchaseUrgency (to use later for sort)
+	// Based number of days we can label each item
+	// Return sorted array: 1) active first inactive last 2) purchaseUrgency (days until next purchase) ascending 3) alphabetical
+
+	shoppingList.forEach((item) => {
+		let daysUntilNextPurchase;
+		let daysSinceLastPurchase;
+
+		if (item.dateNextPurchased) {
+			daysUntilNextPurchase = getDaysBetweenDates(
+				item.dateNextPurchased.toMillis(),
+				Date.now(),
+			);
+			item.purchaseUrgency = daysUntilNextPurchase;
```

Comment:
mapping through an array and modifying it at the same time could be considered a bad practice and may lead to some unexpected behavior. Consider using the `map` method here, instead of `forEach`, which will return a new array and then sort it.
The implementation won't change much, you will just return a new item at the end of each iteration, something like this : 
`return { ...item, purchaseUrgency: daysUntilNextPurchase, urgencyLabel }`

Reply:
You're right. We now have `map` instead of `forEach` and also added a return statement.